pkg/dockerregistry/server/digestcache_test.go

```diff
@@ -0,0 +1,556 @@
+package server
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"k8s.io/kubernetes/pkg/util/diff"
+)
+
+const (
+	allowedDeviation = time.Millisecond * 10
+
+	ttl1m = time.Minute
+	ttl5m = time.Minute * 5
+	ttl8m = time.Minute * 8
+)
+
+var stale = time.Unix(0, 0)
+
+func TestRepositoryBucketAdd(t *testing.T) {
+	now := time.Now()
+
+	generated := make([]bucketEntry, bucketSize)
+	for i := 0; i < bucketSize; i++ {
+		generated[i] = bucketEntry{
+			repository: fmt.Sprintf("gen%d", i),
+			evictOn:    now.Add(ttl5m),
+		}
+	}
+
+	for _, tc := range []struct {
+		name            string
+		ttl             time.Duration
+		repos           []string
+		entries         []bucketEntry
+		expectedEntries []bucketEntry
+	}{
+		{
+			name:  "no existing entries",
+			ttl:   ttl5m,
+			repos: []string{"a", "b"},
+			expectedEntries: []bucketEntry{
+				{
+					repository: "a",
+					evictOn:    now.Add(ttl5m),
```

Comment:
won't these times cause test flakes? `now` won't be the same when this testcase is set up and when the test runs